src/lib/marshal/workflow/writer.ts

```diff
@@ -270,7 +263,7 @@ const buildWorkflowDirBundle = (
       const extractedFilePath = get(
         localWorkflowStepsByRef,
         `${step.ref}.${objPathStr}${FILEPATH_MARKER}`,
-      );
+      ) as string;
```

Comment:
```suggestion
      );
```
I'm not sure if this is right, because it may or may not result in a value.

For making the TS compiler happy, I would probably add an additional "if" line below, like:
```
if (!extractedFilePath && !extractByDefault) continue;
if (typeof extractedFilePath !== "string") continue;
```